shephard-core/src/main/java/com/devsda/platform/shephardcore/resources/ExecutionDocumentResources.java

```diff
@@ -0,0 +1,78 @@
+package com.devsda.platform.shephardcore.resources;
+
+import com.devsda.platform.shephardcore.constants.ShephardConstants;
+import com.devsda.platform.shephardcore.service.ExecuteWorkflowService;
+import com.devsda.platform.shephardcore.service.ExecutionDocumentService;
+import com.devsda.platform.shepherd.constants.ResourceName;
+import com.devsda.platform.shepherd.model.ExecuteWorkflowRequest;
+import com.devsda.utils.httputils.loader.JsonLoader;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.inject.Inject;
+import com.mongodb.MongoWriteException;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.util.JSON;
+import org.bson.Document;
+import org.bson.json.JsonMode;
+import org.bson.json.JsonWriterSettings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.net.InetAddress;
+import java.util.HashMap;
+import java.util.Map;
+
+@Path(ShephardConstants.Resources.EXECUTION_DETAILS)
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class ExecutionDocumentResources {
+
+    @Inject
+    private ExecutionDocumentService executeWorkflowService;
+
+    @Path("/{executionID}")
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response fetchExecutionDetails(@PathParam("executionID") Integer executionID) throws Exception {
+
+        try{
+            Document result =  this.executeWorkflowService.fetchExecutionDetails(executionID);
+            if(result == null) {
+                return Response.status(Response.Status.NOT_FOUND).entity("No Document found for " + executionID).build();
+            }
+
+            ObjectMapper mapper = new ObjectMapper();
+            mapper.configure(com.fasterxml.jackson.databind.SerializationFeature.WRITE_DATES_AS_TIMESTAMPS , false);
+
+            String strigyfiedObject = mapper.writeValueAsString(result);
+            return Response.ok(strigyfiedObject).build();
+        }catch(Exception e) {
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity("Something went wrong " + e.getMessage()).build();
+        }
+    }
+
+    @Path("/{executionID}")
+    @PUT
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response updateExecutionDetails(@PathParam("executionID") Integer executionID,  String updatedExecutionDetails) throws Exception {
+
```

Comment:
We need to add logging.
We can prioritize later once we complete V1 cut pipeline.

Reply:
Sure.